source/admin_manual/mailbox_formats/Maildir.rst

```diff
@@ -0,0 +1,415 @@
+   has been updated to allow more uniqueness identifiers)
+2. Do ``stat(tmp/<filename>)``. If the ``stat()`` found a file, wait 2 seconds
+   and go back to step 1.
+3. Create and write the message to ``tmp/<filename>``.
+4. ``link()`` it into ``new/`` directory. Although not mentioned here, the
+   ``link()`` could again fail if the mail existed in ``new/`` dir. In that
+   case you should probably go back to step 1.
+
+All this trouble is rather pointless. Only the first step is what really
+guarantees that the mails won't get overwritten, the rest just sounds nice.
+Even though they might catch a problem once in a while, they give no
+guaranteed protection and will just as easily pass duplicate filenames through
+and overwrite existing mails.
+
+Step 2 is pointless because there's a race condition between steps 2 and 3.
+PID/host combination by itself should already guarantee that it never finds
+such a file. If it does, something's broken and the ``stat()`` check won't
+help since another process might be doing the same thing at the same time, and
+you end up writing to the same file in ``tmp/``, causing the mail to get
+corrupted.
+
+In step 4 the ``link()`` would fail if an identical file already existed in
+the Maildir, right? Wrong. The file may already have been moved to ``cur/``
+directory, and since it may contain any number of flags by then you can't
+check with a simple ``stat()`` anymore if it exists or not.
+
+Step 2 was pointed out to be useful if clock had moved backwards. However,
+this doesn't give any actual safety guarantees because an identical base
+filename could already exist in ``cur/``. Besidesm if the system was just
+rebooted, the file in ``tmp/`` could probably be even overwritten safely
+(assuming it wasn't already ``link()``\ ed to ``new/``).
+
+So really, all that's important in not getting mails overwritten in your
+Maildir is step 1: Always create filenames that are guaranteed to be unique.
+Forget about the 2 second waits and such that the Qmail's man page talks
+about.
+
+.. _`Qmail's how a message is delivered page`: https://www.qmail.org/man/man5/maildir.html
+
+Maildir and Mail Header Metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Unlike when using :ref:`mbox <mbox_mbox_format>` as
+:ref:`mailbox format <mailbox_formats>`, where mail headers (for example
+``Status``, ``X-UID``, etc.) are used to determine and store metadata, the
+mail headers within Maildir files are (usually) **not** used for this purpose
+by Dovecot; neither when mails are created/moved/etc. via IMAP nor when
+Maildirs are placed (e.g., copied or moved in the filesystem) in a mail
+location (and then "imported" by dovecot).
+
+Therefore, it is (usually) **not** necessary, to strip any such mail headers
+at the MTA, MDA, or LDA (as is recommended with
+:ref:`mbox <mbox_mbox_format>`).
+
+There is one exception, though, namely when ``pop3_reuse_xuidl = yes`` (which
```

Comment:
settings should be references a
```
:ref:`pop3_reuse_xuidl=yes <setting-pop3_reuse_xuidl>`
```

Reply:
ok, although it is annoying that you can't nest formatting so we lose monospace text with this change :(